terraform/games/minecraft/cloudconfig.tf

```diff
@@ -59,7 +59,7 @@ data "template_file" "minecraft" {
                 - /srv/minecraft-spot/data:/data
               environment:
                 EULA: "TRUE"
-                MAX_RAM: "7G"
+                MAX_MEMORY: "7G"
```

Comment:
This should probably be a user-defined variable since the size corresponds with instance type. It would also be nice to be able to add additional environment variables to this section. Terraform has some built in yaml functions that could take a dictionary variable in, merge it with another, and the pass the result through yaml:

```hcl
locals {
  environment_defaults = merge(
    {
      "EULA": "TRUE",
      "MAX_MEMORY": var.max_memory,
      "TYPE": var.modpack_type,
    },
    (var.modpack_type == "FTB" ? {"FTB_SERVER_MOD": var.modpack_version} : {"CF_SERVER_MOD": var.modpack_version}),
  )
}

# later inside the template...
${yamlencode({"environment": merge(local.environment_defaults, var.environment_overrides)})}
```

Thoughts?

https://www.terraform.io/docs/configuration/functions/yamlencode.html
https://www.terraform.io/docs/configuration/functions/merge.html

Reply:
I am already merging environments in my setup, but I'll wait until this PR is done to include these changes in another. Trying to keep this one strictly on bug fixing for now.